### PR TITLE
HADOOP-17904. Test Result Not Working In Jenkins Result.

### PR DESCRIPTION
**Pull request by ayushtkn**
Working Example:
https://github.com/apache/hadoop/pull/2773#issuecomment-916059913

The Test Report link works in the above build

https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2773/10/testReport/

### CODE REVIEW

**Review by virajjasani — APPROVED** [review]
+1 (non-binding), thanks a lot @ayushtkn

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 45s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  shelldocs  |   0m  0s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  12m 53s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  15m 40s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  15m 26s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 35s |  |  The patch does not generate ASF License warnings.  |
|  |   |  47m 38s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3413/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3413 |
| Optional Tests | dupname asflicense codespell shellcheck shelldocs |
| uname | Linux c420542b7dd2 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / fd40d61242ecccadc87d98448e02ec9e97b593ff |
| Max. process+thread count | 685 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3413/1/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by goiri** [comment]
This looks good but I don't have a lot of experience here.
@GauthamBanasandra you have been working around here, can you confirm?

File: dev-support/Jenkinsfile [hunk 1]
```diff
@@ -30,8 +30,9 @@ def publishJUnitResults() {
     boolean surefireReportsExist = findCmdExitCode == 0
     if (surefireReportsExist) {
         echo "XML files found under surefire-reports, running junit"
+        SRC = "${SOURCEDIR}/**/target/surefire-reports/*.xml".replace("$WORKSPACE/","")
```
Comment (GauthamBanasandra) [inline dev-support/Jenkinsfile:34]: Could you please add a comment here mentioning that the path must be relative to the checkout dir?

**Review by GauthamBanasandra — CHANGES_REQUESTED** [review]

File: dev-support/Jenkinsfile (hunk 1 above)
Comment (GauthamBanasandra) [reply dev-support/Jenkinsfile:34]: This path used to have `env.SOURCEDIR` prefix before I made this change - https://github.com/apache/hadoop/blob/ef5dbc7283a9283bc446511c3a49304ad3923037/dev-support/Jenkinsfile#L145. Wasn't `env.SOURCEDIR` resolving to the absolute path back then?

Comment (ayushtkn) [reply dev-support/Jenkinsfile:34]: I don't think so:
https://github.com/apache/hadoop/blob/ef5dbc7283a9283bc446511c3a49304ad3923037/dev-support/Jenkinsfile#L32

Comment (ayushtkn) [reply dev-support/Jenkinsfile:34]: Now:
https://github.com/apache/hadoop/blob/trunk/dev-support/Jenkinsfile#L259

```
                SOURCEDIR = "${WORKSPACE}/ubuntu-focal/src"
```

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  12m 43s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  shadedclient  |  13m 45s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 27s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  shadedclient  |  13m 55s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 34s |  |  The patch does not generate ASF License warnings.  |
|  |   |  44m 59s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3413/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3413 |
| Optional Tests | dupname asflicense codespell shellcheck shelldocs |
| uname | Linux 947f61faaec8 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 7c9d751a905da695e30d83a99db6d03539607ebe |
| Max. process+thread count | 542 (vs. ulimit of 5500) |
| modules | C:  U:  |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3413/2/console |
| versions | git=2.25.1 maven=3.6.3 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Review by iwasakims — APPROVED** [review]
LGTM

**Review by goiri — APPROVED** [review]

**Pull request merged**